core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -489,6 +491,24 @@ class BlockManagerMasterEndpoint(
       storageLevel: StorageLevel,
       memSize: Long,
       diskSize: Long): Boolean = {
+    logInfo(s"Updating block info on master ${blockId} for ${blockManagerId}")
+
+    if (blockId.isInternalShuffle) {
+      blockId match {
+        case ShuffleIndexBlockId(shuffleId, mapId, _) =>
+          // Don't update the map output on just the index block
```

Comment:
I see that `getMigrationBlocks` returns the sub blocks in the order of `[index, data]`, but I wonder if the sub blocks are indeed migrated strictly in order ? Is it ever possible to have a data block updated before an index block ? It seems that we would want both, before we update the mapOutputTracker -- but I am not 100% clear on how that is ensured.

Reply:
Yeah sub blocks are migrated in order (we don't parallel map them).